atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/OracleTableNameGetterImpl.java

```diff
@@ -118,12 +124,37 @@ private Set<TableReference> getTableReferencesFromShortTableNamesWithPrefix(
                 .collect(Collectors.toSet());
     }
 
-    private Set<String> getLongTableNamesFromTableNames(
-            ConnectionSupplier connectionSupplier, Set<String> shortTableNames) throws TableMappingNotFoundException {
+    private Set<String> getLongTableNames(ConnectionSupplier connectionSupplier, Set<String> shortTableNames) {
+        Map<String, String> shortToLongTableNames =
+                oracleTableNameUnmapper.getShortToLongTableNamesFromMappingTable(connectionSupplier, shortTableNames);
+        Set<String> unmappedTableNames = getUnmappedTableNames(shortToLongTableNames, shortTableNames);
+
         if (useTableMapping) {
-            return oracleTableNameUnmapper.getLongTableNamesFromMappingTable(connectionSupplier, shortTableNames);
+            logIfUnmappedTablesPresent(unmappedTableNames);
+            return new HashSet<>(shortToLongTableNames.values());
+        }
+
+        return unmappedTableNames;
+    }
+
+    private Set<String> getUnmappedTableNames(
+            Map<String, String> shortToLongTableNameMapping, Set<String> expectedShortTableNames) {
+        Set<String> mappedShortTableNames = shortToLongTableNameMapping.keySet().stream()
+                .map(tableName -> tableName.toLowerCase(Locale.ROOT))
```

Comment:
Alternative was TreeSet with case insensitive ordering, but we use lowercase matching in our sql queries (and it's a little cheaper to do it this way)